ProcessMaker/Templates/ScreenTemplate.php

```diff
@@ -268,25 +277,66 @@ protected function getRequestInclude(Request $request) : array
     }
 
     /**
-     * Check if an existing process template with the same name exists.
-     * If exists, return an array with the existing template ID and name.
-     * Otherwise, return null.
-     * @param Request $request
+     * Create a new screen template.
      *
-     * @return array|null Array containing the existing template ID and name or null if no existing template found
+     * @param  array  $data  The data for creating the screen template
+     * @param  array  $payload  The payload for the screen template
+     * @return \App\Models\ScreenTemplates  The created screen template
      */
-    public function existingTemplate($request) : ?array
+    protected function createScreenTemplate(array $data, array $payload) : ScreenTemplates
     {
-        $templateId = $request->id;
-        $name = $request->name;
+        $screenType = Screen::findOrFail($data['asset_id'])->type;
+        $screenTemplate = ScreenTemplates::make($data)->fill([
+            'manifest' => json_encode($payload),
+            'user_id' => auth()->id(),
+            'screen_type' => $screenType,
+            'media_collection' => '',
+        ]);
+        $screenTemplate->saveOrFail();
+        $screenTemplate->media_collection = 'st-' . $screenTemplate->uuid . '-media';
+        $screenTemplate->saveOrFail();
 
-        $template = ScreenTemplates::where(['name' => $name])->where('id', '!=', $templateId)->first();
-        if ($template !== null) {
-            // If same asset has been Saved as Template previously,
-            // offer to choose between “Update Template” and “Save as New Template”
-            return ['id' => $template->id, 'name' => $name];
+        return $screenTemplate;
+    }
+
+    private function updateScreenTemplateData(array $data, array $payload)
+    {
+        ScreenTemplates::where('name', $data['name'])->update([
+            'description' => $data['description'],
+            'is_public' => $data['make_public'] ? 1 : 0,
```

Comment:
This seems to be working fine; but when I created a template without clicking on the "Make public" checkbox, my template appeared in the Public Templates section tab... I believe it should only appear under "My Templates"

maybe, this issue could be addressed in a separate ticket?

Reply:
@eiresendez Filtering of the My template and public templates will be implemented in another ticket